neuralmonkey/runners/runner.py

```diff
@@ -1,50 +1,69 @@
-            extra_fetches = []
-
-        loss_with_gt_ins = 0.0
-        loss_with_decoded_ins = 0.0
-        batch_count = 0
-        for batch in batched_dataset:
-            decoded = self.decoder.decoded
-            vocab = self.decoder.vocabulary
-            batch_feed_dict = feed_dicts(batch, coders, train=False)
-            batch_count += 1
-
-            # if is a target sentence, compute also the losses
-            # otherwise, just compute zero
-            if dataset.has_series(self.decoder.data_id):
-                losses = [self.decoder.train_loss,
-                          self.decoder.runtime_loss]
-            else:
-                losses = [tf.zeros([]), tf.zeros([])]
-
-            (loss_with_gt_ins, loss_with_decoded_ins), \
-                    decoded, fetches_batch = \
-                    sess.run((losses, self.decoder.decoded, extra_fetches),
-                             feed_dict=batch_feed_dict)
-            decoded_sentences_batch = \
-                    vocab.vectors_to_sentences(decoded)
-
-            decoded_sentences += decoded_sentences_batch
-            evaluated_fetches += [fetches_batch]
-
-        return decoded_sentences, \
-               loss_with_gt_ins / batch_count, \
-               loss_with_decoded_ins / batch_count, \
-               evaluated_fetches
+from neuralmonkey.tf_manager import RunResult
+from neuralmonkey.runners.base_runner import BaseRunner, \
+    Executable, ExecutionResult, NextExecute
+
+# tests: mypy,pylint
+# pylint: disable=too-few-public-methods
+
+
+class GreedyRunner(BaseRunner):
+
+    def __init__(self, output_series, decoder):
+        super(GreedyRunner, self).__init__(output_series, decoder)
+
+    def get_executable(self, train=False, summaries=True):
+        if train:
+            losses = [self._decoder.train_loss,
+                      self._decoder.runtime_loss]
+        else:
+            losses = [tf.zeros([]), tf.zeros([])]
+        to_run = losses + self._decoder.runtime_logprobs
+        return GreedyRunExecutable(self.all_coders, to_run,
+                                   self._decoder.vocabulary)
+
+    @property
+    def loss_names(self) -> List[str]:
+        return ["train_xent", "runtime_xent"]
+
+
+class GreedyRunExecutable(Executable):
+
+    def __init__(self, all_coders, to_run, vocabulary):
+        self.all_coders = all_coders
+        self.to_run = to_run
+        self.vocabulary = vocabulary
+
+        self.decoded_sentences = []
+        self.result = None  # type: Option[ExecutionResult]
+
+    def next_to_execute(self) -> NextExecute:
+        """Get the feedables and tensors to run."""
+        return self.all_coders, self.to_run, {}
+
+    def collect_results(self, results: List[List[RunResult]]) -> None:
+        train_loss = 0.
+        runtime_loss = 0.
+        summed_logprobs = [-np.inf for _ in range(len(self.to_run) - 2)]
+
+        for sess_result in results:
+            train_loss += sess_result[0]
```

Comment:
Jestli se ptáš obecně v Pythonu, tak jsem teď zkusil napsat 

```
a = {"b": {"foo": 5}}
print(a["b"]["foo"])
```

A napsalo to 5